internal/github_client/github_client.go

```diff
@@ -78,3 +73,78 @@ func InitConfig() GithubClient {
 	return auth
 
 }
+
+// Helper function to allow testing client initialization with custom http clients
+func initTokenClient(c *TokenConfig, httpClient *http.Client) *github.Client {
+	if httpClient == http.DefaultClient {
+		ctx := context.Background()
+		ts := oauth2.StaticTokenSource(
+			&oauth2.Token{AccessToken: c.Token},
+		)
+		httpClient = oauth2.NewClient(ctx, ts)
+	}
+	return github.NewClient(httpClient)
+}
+
+// Helper function to allow testing client initialization with custom http clients
```

Comment:
nit: While they are not perfect, for such cases you could also go with functional options. Not necessary here but perhaps something to consider for the next time you deal with optional options 🙂

Reply:
I see what you mean, however I feel like there has been a little of a scope creep on this PR, this wasn't supposed to be about updating packages or changing how functions are implemented, the focus here is to add the ability of authenticating without InstallationID using OrgName and RepoName, something like changing the functions to use functional options feels like would fit better in a future PR to refactor.